qemu/tests/seabios_reboot_timeout.py

```diff
@@ -0,0 +1,50 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM seabios test:
+    1) Start VM
+    2) Check if reboot-timeout option works as expected
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_output(session_obj):
+        """
+        Use the function to short the lines in the scripts
+        """
+        return session_obj.get_stripped_output()
+
+    def reboot_timeout_check():
+        """
+        reboot-timeout check
+        """
+        return re.search(pattern, get_output(seabios_session), re.S)
+
+    error_context.context("Start VM", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    timeout = float(params.get("login_timeout", 360))
+    seabios_session = vm.logsessions['seabios']
+    rb_timeout = int(params["boot_reboot_timeout"])
+    if rb_timeout < 0:
+        raise NotImplementedError()
+    elif rb_timeout > 65535:
+        pattern = "No bootable device.*Retrying in 65 seconds"
+    else:
+        pattern = "No bootable device.*Retrying in %d seconds" % (rb_timeout//1000)
```

Comment:
line 42 and line 44 will give the same output no matter rb_timeout > 65535 or not.
if rb_timeout = 65536, let's presume it goes to line 44, rb_timeout/1000  = 65 seconds. or it goes to line 42, also 65 seconds.
so line 41 and 42 can be removed.

However, you can leave some log to give users a hint that 65535 is the limitation. Up to you.

Reply:
＠ybduan You are right, it need to retain >65535 scenario.  once reboot_timeout value >65535. qemu-kvm will force it to 65535.

Reply:
then how about this:
...
else rb_timeout > 65535:
    rb_timeout = 65535
pattern = "No bootable device. *Retrying in %d seconds" % (rb_timeout//1000)

Does the code sounds like what you expected? @xiangchunfu